main/src/cgeo/geocaching/CacheDetailActivity.java

```diff
@@ -1591,6 +1592,28 @@ private void updateChirpWolfBox() {
                 });
             }
         }
+
+        private void updateALCBox() {
+            final boolean isEnabled = cache.getType() == CacheType.ADVLAB && StringUtils.isNotEmpty(cache.getUrl());
+            final Intent alc = ProcessUtils.getLaunchIntent(getString(R.string.package_alc));
+            final LinearLayout boxCW = view.findViewById(R.id.alc_box);
+            boxCW.setVisibility(isEnabled ? View.VISIBLE : View.GONE);
+            final TextView msgCW = view.findViewById(R.id.alc_text);
+            msgCW.setText(null != alc ? R.string.cache_alc_start : R.string.cache_alc_install);
+            if (isEnabled) {
+                final ImageButton buttonALC = view.findViewById(R.id.send_to_alc);
+                buttonALC.setOnClickListener(v -> {
+                    // re-check installation state, might have changed since creating the view
+                    if (null != alc) {
```

Comment:
Strictly speaking you should do an actual reevaluation whether ALC package is still present (so do a new call to `ProcessUtils.getLaunchIntent`), as the user might have uninstalled it meanwhile. A null check for `alc` won't work here.
OTOH it's not very likely, that a user opens a cache detail page, uninstalls ALC, and the taps the button, so we can probably remove the `if (null != alc)` check completely.

Reply:
Ignore my previous comment :-(
Have just realized that you also offer a link to the app store. For this you still need the null check.